Polyhedron/demo/Polyhedron/Scene_polygon_soup_item.cpp

```diff
@@ -420,9 +420,23 @@ void Scene_polygon_soup_item::inside_out()
 }
 
 bool
-Scene_polygon_soup_item::orient()
+Scene_polygon_soup_item::orient(std::vector<std::pair<std::size_t, std::size_t> >& non_manifold_vertices)
 {
+  struct Visitor : public CGAL::Polygon_mesh_processing::internal::Polygon_soup_orientation_visitor
+  {
+    std::vector<std::pair<std::size_t, std::size_t> >& nm_vertices;
```

Comment:
does the demo require a pair?

Reply:
Yes. To separate NM vertices from NM Edges endpoints, you have to check if a nm vertex is one of the endpoints of an edge. But you cannot know which vertex you are comparing, as it has been duplicated, so you have to compare with both, hence the pair.

Reply:
can't you simply insert both in your container?

Reply:
If one of the members of the pair matches, both members need to be removed, so I need the pair